pkg/apis/core/validation/validation.go

```diff
@@ -4162,28 +4171,74 @@ func validateConfigMapNodeConfigSourceSpec(source *core.ConfigMapNodeConfigSourc
 	return append(allErrs, validateConfigMapNodeConfigSource(source, fldPath)...)
 }
 
+// validation specififc to Node.Status.Config
+func validateNodeConfigStatus(status *core.NodeConfigStatus, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if status.Assigned != nil {
+		// Assigned is a reflection of the spec, which acts as an ACK from the Kubelet regarding recognition,
+		// of the assignment. As such, it is validated against the same rules as the spec, rather than status.
+		allErrs = append(allErrs, validateNodeConfigSourceSpec(status.Assigned, fldPath.Child("assigned"))...)
+	}
+	if status.Active != nil {
+		allErrs = append(allErrs, validateNodeConfigSourceStatus(status.Active, fldPath.Child("active"))...)
+	}
+	if status.LastKnownGood != nil {
+		allErrs = append(allErrs, validateNodeConfigSourceStatus(status.LastKnownGood, fldPath.Child("lastKnownGood"))...)
+	}
+	return allErrs
+}
+
+// validation specific to Node.Status.Config.(Active|Assigned|LastKnownGood)
+func validateNodeConfigSourceStatus(source *core.NodeConfigSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	count := int(0)
+	if source.ConfigMap != nil {
+		count++
+		allErrs = append(allErrs, validateConfigMapNodeConfigSourceStatus(source.ConfigMap, fldPath.Child("configMap"))...)
+	}
+	// add more subfields here in the future as they are added to NodeConfigSource
+
+	// exactly one reference subfield must be non-nil
+	if count != 1 {
+		allErrs = append(allErrs, field.Invalid(fldPath, source, "exactly one reference subfield must be non-nil"))
+	}
+	return allErrs
+}
+
+// validation specific to Node.Status.Config.(Active|Assigned|LastKnownGood).ConfigMap
+func validateConfigMapNodeConfigSourceStatus(source *core.ConfigMapNodeConfigSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if string(source.UID) == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("uid"), "uid must be set in status"))
+	}
+	// TODO(#63221): require ResourceVersion in status when we start respecting ConfigMap mutations (the Kubelet isn't tracking it internally until
+	// that PR, which makes it difficult to report for now).
```

Comment:
even if we don't track mutations, we should still have the actual configmap in hand when we report status, right? why can't we report the resourceVersion we're using?

Reply:
We set status when we bootstrap the controller, when an active config becomes the new last-known-good, and when we ack the assigned config. The last of those three doesn't need UID/ResourceVersion because it's a reflection of spec; but during the former two we only have metadata from SerializedNodeConfigSources. Until I made the changes in #63221 to respect ConfigMap updates, we weren't setting ResourceVersion in the SerializedNodeConfigSources (UID was the only identifier that mattered for resolving a checkpoint), so we didn't know it at bootstrap/lkg-update time. 

It wouldn't be too hard to port that fix back to this PR, though.